### PR TITLE
C++ API remove mgp::memory usage

### PR DESCRIPTION
**Pull request by Ignition**
### Description

Change wording to change the focus around mgp::memory. It was softly deprecated before, but now we want to encourage modules to be updated.

### Pull request type

Please check what kind of PR this is:

- [x] Fix or improvement of an existing page
- [ ] New documentation page, release related

### Related PRs and issues

PR this doc page is related to: 
memgraph/memgraph#2424

Closes:
(paste the link to the issue it closes)

### Checklist:

- [x] Check all content with Grammarly
- [x] Perform a self-review of my code
- [ ] Make corresponding changes to the rest of the documentation (consult with the DX team)
- [ ] The build passes locally
- [ ] My changes generate no new warnings or errors
- [ ] Add a corresponding label
- [ ] If release-related, add a product and version label
- [ ] If release-related, add release note on product PR

### CODE REVIEW

File: pages/custom-query-modules/cpp/cpp-api.md [hunk 1]
```diff
@@ -22,7 +22,7 @@ Memgraph in order to use them.
 With this API it’s possible to extend your Cypher queries with **functions** and **procedures** with
 `AddProcedure` and `AddFunction`.
 
-The API needs memory access to add procedures and functions, this can be done with either `mgp::MemoryDispatcherGuard guard(memory);` or `mgp::memory = memory;`, where the use of the former is advised since `mgp::memory = memory;` is not thread-safe and will be deprecated in the future.
+The API needs memory access to add procedures and functions, this can be done with `mgp::MemoryDispatcherGuard guard(memory);`. Old code used `mgp::memory = memory;`, but this is not thread-safe and has been deprecated. v2.18.1 onwards you should modify your C++ modules and recompile.
```
Comment (hal-eisen-MG) [inline pages/custom-query-modules/cpp/cpp-api.md:25]: Should or must?

File: pages/custom-query-modules/cpp/cpp-example.mdx [hunk 2]
```diff
@@ -392,13 +392,13 @@ void RandomWalk(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result, m
...
+`mgp::memory` was a global object, that meant all of the procedures and
 functions in a single shared library will refer to the same `mgp::memory`
 object. As a result, calling such callables simultaneously from multiple threads
 will lead to incorrect memory usage. This also includes the case when the same
-callable is called from different user sessions. This is a constraint when using
-`mgp::memory` so the use of the thread-safe `MemoryDispatcherGuard guard(memory)`
-is advised instead.
+callable is called from different user sessions. For some versions of memgraph 
+this was deprecated and the use of the thread-safe `MemoryDispatcherGuard guard(memory)`
+is advised instead. Its has now been removed and C++ modules should be recompiled for v2.18.1+.
```
Comment (hal-eisen-MG) [inline pages/custom-query-modules/cpp/cpp-example.mdx:401]: Should or must?

File: pages/custom-query-modules/cpp/cpp-api.md (hunk 1 above)
Comment (andrejtonev) [reply pages/custom-query-modules/cpp/cpp-api.md:25]: v2.18.1 still allows this, but the user should be careful and not execute procedures that share the memory in parallel.
v2.21 setting mgp::memory will cause a compilation error.

Comment (hal-eisen-MG) [reply pages/custom-query-modules/cpp/cpp-api.md:25]: Then please change the language in this PR.

**Comment by hal-eisen-MG** [comment]
Changed PR to target the `memgraph-2-21` branch instead of `main`.

File: pages/custom-query-modules/cpp/cpp-api.md (hunk 1 above)
Comment (gitbuda) [reply pages/custom-query-modules/cpp/cpp-api.md:25]: Added the required note about v2.21.

File: pages/custom-query-modules/cpp/cpp-example.mdx (hunk 2 above)
Comment (gitbuda) [reply pages/custom-query-modules/cpp/cpp-example.mdx:401]: Added the required note about v2.21.

**Pull request merged**